TEST-8

KyKyMB3R opened the pull request.

KyKyMB3R committed:
Fix all bags found by testers

GreenBudgie commented:
В заголовке нет номера задачи:

> В заголовок написать номер задачи (TEST-8) и добавить краткое описание с изменениями.

GreenBudgie commented on src/Waypoint.java:
```diff
@@ -5,7 +5,13 @@ public record Waypoint(int x, int y) {
+        return Math.sqrt((dx * dx) + (dy * dy));
+    }
+
+    public double distanceToZeroCord(Waypoint otherWaypoint) {
```
Зачем нужна эта функция?

KyKyMB3R replied:
Чтобы считать расстояние от нулевой точки до нашей точки

GreenBudgie replied:
А разве это не то же самое, что и `distanceTo(new Waypoint(0, 0))`?

GreenBudgie replied:
У тебя эта функция не использует поля текущего объекта, она работает только с `otherWaypoint`

GreenBudgie commented:
А еще b**a**g - это сумка, а b**u**g - это баг

GreenBudgie commented on src/Tester.java:
```diff
@@ -1,7 +1,51 @@
+        Path test = new Path();
+        test.addWaypoint(3, 1);
+        test.addWaypoint(5, 2);
+        double expectedResult = 2.23606797749979;
```
Это некорректный ожидаемый результат. Я предполагаю, что ты тестировал очень крутым способом:
1. Запустить
2. Посмотреть на результат
3. Сделать этот результат ожидаемым значением

GreenBudgie commented on src/Tester.java:
```diff
@@ -1,7 +1,51 @@
+        double expectedResult = 2.23606797749979;
+        double testResult = test.getLength();
+        if (testResult != expectedResult) {
+            throw new Exception("Exception in getLengthZeroToWaypoint (test 1)");
```
Давай писать более подробные ошибки, когда тесты падают. Например, выводить конкретную информацию о том, какое было ожидаемое значение и какое получилось актуальное.

Этот коммент относится ко всем исключениям в тестах.

KyKyMB3R committed:
Tester - new 5 tests
Path - fix getLength, create getWaypoints, create object Waypoint in addWaypoint.
Waypoint - fix distanceTo

KyKyMB3R committed:
Tester - new 5 tests
Path - fix getLength, create getWaypoints, create object Waypoint in addWaypoint.
Waypoint - fix distanceTo
